lib/charms/layer/azure.py

```diff
@@ -202,11 +215,11 @@ def _get_tenant_id(subscription_id):
         return match.group(1)
 
 
-def _azure(cmd, subcmd, *args, return_stderr=False):
+def _azure(cmd, *args, return_stderr=False):
     """
     Call the azure-cli tool.
     """
-    cmd = ['az', '--output=json', cmd, subcmd]
+    cmd = ['az', cmd]
```

Comment:
Not all `az` commands have a sub-command, and the `--output` option has to come after the sub-command, if present, but defaults to `json` anyway, so it's easier to just remove it.